### PR TITLE
Cleanup Keystone contracts

### PR DESCRIPTION
**Pull request by DeividasK**
Changes:
- Remove `require` in favor of custom errors.
- Change returns to be explicit.
- Update the compile script to work with Foundry. For some reason, `solc` did not work with a library reference. Given that we are using Foundry anyways, it made more sense to update the contract compilation script instead of trying to make it work with `solc`.

### CODE REVIEW

**Comment by archseer** [comment]
> * Inline Utils inside the contract. Copy and paste the split report into a mock contract.

I've kept it separate since the report parsing code should be a library for downstream consumers (being able to split out metadata vs report). But maybe this is more efficient for inlining?

File: contracts/src/v0.8/keystone/KeystoneForwarder.sol [hunk 1]
```diff
@@ -4,77 +4,135 @@ pragma solidity ^0.8.19;
 import {IForwarder} from "./interfaces/IForwarder.sol";
 import {ConfirmedOwner} from "../shared/access/ConfirmedOwner.sol";
 import {TypeAndVersionInterface} from "../interfaces/TypeAndVersionInterface.sol";
-import {Utils} from "./libraries/Utils.sol";
 
-// solhint-disable gas-custom-errors, no-unused-vars
 contract KeystoneForwarder is IForwarder, ConfirmedOwner, TypeAndVersionInterface {
```
Comment (cds95) [inline contracts/src/v0.8/keystone/KeystoneForwarder.sol:8]: What is the KeystoneForwarder?

File: contracts/src/v0.8/keystone/interfaces/IForwarder.sol [hunk 2]
```diff
@@ -1,5 +0,0 @@
-// SPDX-License-Identifier: MIT
-pragma solidity ^0.8.19;
-
-/// @title IForwarder - forwards keystone reports to a target
-interface IForwarder {}
```
Comment (DeividasK) [inline contracts/src/v0.8/keystone/interfaces/IForwarder.sol:5]: Removed it as it is empty. We can add it later if needed.

**Comment by DeividasK** [comment]
> > * Inline Utils inside the contract. Copy and paste the split report into a mock contract.
> 
> I've kept it separate since the report parsing code should be a library for downstream consumers (being able to split out metadata vs report). But maybe this is more efficient for inlining?

It is more efficient to inline, but I like having a library for consumers. I renamed it from Utils and added a comment about it.

**Pull request closed without merging**